src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,128 @@
+//
+// ImageIO - CGImageAnimation.cs
+//
+// Authors:
+//	Whitney Schmidt  <whschm@microsoft.com>
+//
+// Copyright 2020, Microsoft Corp.
+//
+
+using System;
+using System.Runtime.InteropServices;
+using CoreFoundation;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace ImageIO {
+
+    public partial class CGImageAnimation {
+
+        public delegate void CGImageSourceAnimationBlock (nint arg0, CGImage arg1, out bool done /*check*/ );
+
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern unsafe int CGAnimateImageAtURLWithBlock (/* CFURLRef */ IntPtr url, /* CFDictionaryRef _iio_Nullable */ IntPtr options, /* CGImageSourceAnimationBlock */ IntPtr block);
+
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern int CGAnimateImageDataWithBlock ( /* CFDataRef _Nonnull */ IntPtr data, /* CFDictionaryRef _Nullable */ IntPtr options, /* CGImageSourceAnimationBlock _Nonnull */ IntPtr block);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+        public virtual int CGAnimateImage (NSUrl url, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block) {
+            if (url == null)
+                throw new ArgumentNullException ("url");
+            if (block == null)
+                throw new ArgumentNullException ("block");
+
+            int ret;
+            unsafe {
+                BlockLiteral * block_ptr_block;
+                BlockLiteral block_block;
+                block_block = new BlockLiteral ();
+                block_ptr_block = & block_block;
+                block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+                try {
+                    ret = CGAnimateImageAtURLWithBlock (url.Handle, options == null ? IntPtr.Zero : options.Handle, (IntPtr) block_ptr_block);
+                } finally {
+                    block_ptr_block -> CleanupBlock ();
+                }
+            }
+
+            return ret;
+        }
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+        public virtual int CGAnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block) {
```

Comment:
Same: non-virtual and remove the `CG` prefix:

```suggestion
        public int AnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block) {
```